lib/srv/forward/sshserver.go

```diff
@@ -1086,64 +1006,45 @@ func (s *Server) handleX11Forward(ctx context.Context, ch ssh.Channel, req *ssh.
 		},
 	}
 
-	// check if RBAC permits X11 forwarding
-	if !scx.Identity.RoleSet.PermitX11Forwarding() {
-		event.Metadata.Code = events.X11ForwardFailureCode
-		event.Status.Success = false
-		event.Status.Error = "x11 forwarding not permitted"
-		if err := s.EmitAuditEvent(s.closeContext, &event); err != nil {
+	defer func() {
+		if err := s.EmitAuditEvent(ctx, event); err != nil {
```

Comment:
So this doesn't emit until the request has been forwarded?

Is there any potential for abuse here? Could someone activate X11 forwarding, do some things, and then get Teleport to crash before any event is emitted, covering their tracks?

Reply:
`handleX11Forward` isn't called for every xserver request sent by the client (`xeyes`), it's only called on initial setup of the session, when the client sends the `x11-req` request during `tsh ssh - X`.

The actual x11 forwarding will be carried out by the goroutine spawned by `handleX11Forward`, and that goroutine won't start processing xserver requests until the session is fully set up. 

So in short, the event will always be emitted before x11 requests can be forwarded.